src/workloads/query/MatchWithLargeExpression.yml

```diff
@@ -0,0 +1,150 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of expression search for parameter re-use during parameterization. 
+  Before September 2023, parameter re-use had O(n^2) complexity due to using a vector for looking up equivalent expressions. 
+  SERVER-79092 fixed this issue by switching over to a map once the amount of expressions reaches a threshold (currently 50).
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 7
+  DocumentCount: &DocumentCount 10
```

Comment:
This is a very small number of document? Should we be testing a more realistic volume of data?

Reply:
This is specifically geared towards testing a case where the slowness is caused by the size of the query rather than the size of the collection I think- that being said, I'd be in favor of doing something like having two runs: one with a small collection size, and one with a larger collection.

Reply:
Sounds good! Is it worth mentioning that in the description for this workload?